src/components/EnterpriseList/index.jsx

```diff
@@ -3,178 +3,96 @@ import PropTypes from 'prop-types';
-
-    if (enterprises !== prevProps.enterprises) {
-      this.setState({ // eslint-disable-line react/no-did-update-set-state
-        enterprises: enterprises && enterprises.results,
-        pageCount: enterprises && enterprises.num_pages,
-      });
-    }
-    if (searchQuery !== prevProps.searchQuery) {
-      const options = this.formatQueryParams({ search: searchQuery });
-      this.props.history.push(`?${qs.stringify(options)}`);
-      this.getEnterpriseList(options);
-    }
-  }
-
-  getEnterpriseList(options) {
-    this.props.getEnterpriseList(options);
-  }
-
-  clearPortalConfiguration() {
     this.props.clearPortalConfiguration();
-  }
-
-  formatQueryParams(options) {
-    return omitBy({
-      search: options.search,
-    }, isEmpty);
-  }
-
-  formatEnterpriseData(enterprises) {
-    if (!enterprises) {
-      return null;
-    } else if (!enterprises.length) {
-      return [];
-    }
-
-    return enterprises.map(enterprise => ({
-      link: <Link to={`/${enterprise.slug}/admin`}>{enterprise.name}</Link>,
-      name: enterprise.name,
-      slug: enterprise.slug,
-      uuid: enterprise.uuid,
-    }));
+    this.props.getLocalUser();
   }
 
   handleSearch(query) {
-    this.props.setSearchQuery(query);
     this.setState({
       searchQuery: query,
       searchSubmitted: true,
     });
+    this.props.searchEnterpriseList({
+      // Ternary used to turn empty string ('') into undefined so it does not get set in qs
+      search: query ? query : undefined, // eslint-disable-line no-unneeded-ternary
+    });
   }
 
   handleClear() {
-    this.props.setSearchQuery('');
     this.setState({
       searchQuery: '',
+      searchSubmitted: true,
     });
   }
 
-  shouldRenderRedirectToEnterpriseAdminPage() {
-    const { loading, error } = this.props;
-    const { enterprises, pageCount, searchSubmitted } = this.state;
-    return (
-      !loading && !error && enterprises && enterprises.length === 1 &&
-        pageCount === 1 && !searchSubmitted
-    );
-  }
+  formatEnterpriseData = enterprises => enterprises.map(enterprise => ({
+    link: <Link to={`/${enterprise.slug}/admin`}>{enterprise.name}</Link>,
+    name: enterprise.name,
+    slug: enterprise.slug,
+    uuid: enterprise.uuid,
+  }));
 
-  renderEmptyEnterpriseListMessage() {
-    const { searchQuery } = this.props;
-    let emptyMessage = 'You do not have permission to view any enterprise customer data.';
-    if (searchQuery) {
-      emptyMessage = 'There are no enterprise customers that match your search.';
-    }
-    return (
-      <StatusAlert
-        alertType="warning"
-        iconClassName={['fa', 'fa-exclamation-circle']}
-        message={emptyMessage}
-      />
-    );
+  fetchEnterprisesWithSearch = (options) => {
+    const optionsWithSearch = {
+      ...options,
+      // Ternary used to turn empty string ('') into undefined so it does not get set in qs
+      search: this.state.searchQuery ? this.state.searchQuery : undefined,
```

Comment:
Could this actually be `search: this.state.searchQuery || undefined`, to avoid the eslint disable?